### PR TITLE
Fixes #36083 - Remove remote_execution_by_default setting

### PR DESCRIPTION
**Pull request by jeremylenz**
#### What are the changes introduced in this pull request?

* Remove the setting 'Use remote execution by default'
* Everywhere in the code that used it, change behavior to use remote execution no matter what

#### Considerations taken when implementing this change?

Should probably wait to merge this one until any UI changes that depend on it are merged.

#### What are the testing steps for this pull request?

- Test incremental updates via REX and make sure nothing is broken
- Test legacy content host UI and ensure it always uses REX and nothing is broken
- Test new host details page and ensure it always uses REX and nothing is broken
- on new host details page > Content > Packages > kebab > Install packages, ensure 'via Katello agent' option is NOT listed

### CODE REVIEW

File: lib/katello/engine.rb [hunk 1]
```diff
@@ -247,7 +247,7 @@ def self.with_katello_agent?
   end
 
   def self.remote_execution_by_default?
```
Comment (lfu) [inline lib/katello/engine.rb:249]: Is this going to be removed by another PR?

Comment (jeremylenz) [reply lib/katello/engine.rb:249]: No. Maybe I should remove it here..

File: app/lib/actions/katello/content_view/incremental_updates.rb [hunk 2]
```diff
@@ -4,8 +4,8 @@ module ContentView
       class IncrementalUpdates < Actions::EntryAction
         include Helpers::Presenter
 
-        def plan(version_environments, composite_version_environments, content, dep_solve, hosts, description,
-                use_remote_execution = false)
+        def plan(version_environments, composite_version_environments, content, dep_solve, hosts, description) # rubocop:disable Metrics/MethodLength
+          use_remote_execution = true # TODO: remove this when we remove katello-agent dynflow actions
```
Comment (lfu) [inline app/lib/actions/katello/content_view/incremental_updates.rb:8]: This makes code on line 38-40 would never run. 
Do we really need it?

Comment (jeremylenz) [reply app/lib/actions/katello/content_view/incremental_updates.rb:8]: I didn't want to get too deep into editing Dynflow actions here, since Dynflow actions will be in another PR.

**Comment by jeremylenz** [comment]
@lfu Removed `remote_execution_by_default?` method and the single reference to it.

File: engines/bastion_katello/lib/bastion_katello/engine.rb [hunk 3]
```diff
@@ -43,7 +43,6 @@ class Engine < ::Rails::Engine
             'katelloAgentPresent' => ::Katello.with_katello_agent?,
             'remoteExecutionPresent' => ::Katello.with_remote_execution?,
             'hostToolingEnabled' => (::Katello.with_katello_agent? || ::Katello.with_remote_execution?) ? true : false,
-            'remoteExecutionByDefault' => ::Katello.remote_execution_by_default?
```
Comment (lfu) [inline engines/bastion_katello/lib/bastion_katello/engine.rb:46]: `remoteExecutionByDefault` is used by angular code. Maybe this line should be removed by the PR that cleans up angular code.

For this one, we can keep it as:
```
'remoteExecutionByDefault' => true
```

Comment (jeremylenz) [reply engines/bastion_katello/lib/bastion_katello/engine.rb:46]: @chris1984 If you can move this change to your PR, I'll remove it here.

Comment (chris1984) [reply engines/bastion_katello/lib/bastion_katello/engine.rb:46]: :+1:

Comment (jeremylenz) [reply engines/bastion_katello/lib/bastion_katello/engine.rb:46]: actually, If I remove this line but keep the method removed, it will break all the tests on this PR. @lfu I plan to merge the Angular PR first. I'm thinking it may be easier to just keep this PR the way it is..   @chris1984

Comment (chris1984) [reply engines/bastion_katello/lib/bastion_katello/engine.rb:46]: Lol that's fine, I'll take it out of mine

**Comment by jeremylenz** [comment]
fixed rubocop

File: engines/bastion_katello/lib/bastion_katello/engine.rb [hunk 4]
```diff
@@ -42,8 +42,7 @@ class Engine < ::Rails::Engine
             'defaultDownloadPolicy' => !Foreman.in_rake? && db_migrated && Setting['default_download_policy'],
             'katelloAgentPresent' => ::Katello.with_katello_agent?,
             'remoteExecutionPresent' => ::Katello.with_remote_execution?,
-            'hostToolingEnabled' => (::Katello.with_katello_agent? || ::Katello.with_remote_execution?) ? true : false,
-            'remoteExecutionByDefault' => ::Katello.remote_execution_by_default?
+            'hostToolingEnabled' => (::Katello.with_katello_agent? || ::Katello.with_remote_execution?) ? true : false
```
Comment (lfu) [inline engines/bastion_katello/lib/bastion_katello/engine.rb:45]: This line should be taken care of as well by some PR.

**Comment by lfu** [comment]
Tests went well with hammer incremental update.
Content - Errata - apply worked.
Nothing was broken in Content Host and new Hosts detail pages for REX installation.

**Review by lfu — APPROVED** [review]
LGTM 👍🏻
